Tài liệu API cho việc quản lý nhân viên

huubien opened the pull request:
Tài liệu API cho việc quản lý nhân viên có các phương thức Get,Post,Put,Delete
1. Thông tin chi tiết một nhân viên
1.  Tìm kiếm nhân viên
1.  Tạo mới nhân viên
1.  Cập nhập nhân viên
1. Xóa một nhân viên

lannt-xyz committed:
Update interface-document.md

huubien committed:
Update interface-document

Edit description,change body.

huubien committed:
Update interface-document

change body,edit description

huubien committed:
Update interface-document.md

add body, repair Description

huubien committed:
Update interface-document

repair Request Body

huubien committed:
Update interface-document

Repair description.

lanntxyz commented on employee-management/interface-document.md:
```diff
@@ -1,4 +1,4 @@
-# Employee Manage Api Interface specification
+# Employee Manage API Interface specification
 
 ## Table of contents
```
test

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -82,23 +81,44 @@
 |:----|:----|:----|:----|
-|400|invalid_id|Employee ID is invalid|The requested Employee ID is contains Non-Nummeric characters. |
+|400|invalid_employee_id|Employee ID is invalid.|The requested Employee ID is contains Non-Nummeric characters.|
+|400|invalid_employee_id_lenght|Character too long.|Request Employee ID too long.|
```
trường hợp quá ngắn thì sao

huubien replied:
Em đã sửa lại. Sau khi suy nghĩ trường hợp invalid_employee_id_lenght không cần thiết vì mình chỉ cần kiểm tra không hợp lệ hoặc không tồn tại.

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -82,23 +81,44 @@
+
 ## Search Employee API
 
+- Returns the employee information,which corresponding to the condition was described in the [Query paramenters](#query-parameters)
```
coi lại chổ này sai cái gì

huubien replied:
Sau dấu phẩy là space ạ. Em đã chỉnh sửa lại

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -82,23 +81,44 @@
 ## Search Employee API
 
+- Returns the employee information,which corresponding to the condition was described in the [Query paramenters](#query-parameters)
+- Returns the list of employee information which ascending by employee_id
```
`ascending` là thể loại từ gì vậy

`employee_id` -> `Employee ID`

huubien replied:
ascending là động từ ạ. Em sửa lại: Sort the list of employee information result by ascending Employee ID có đúng ko ạ ?

huubien replied:
Em sẽ sửa lại Employee ID

lannt-xyz replied:
`Sort the list of employee information result by ascending`
câu thì đúng, nhưng ko phù hợp với ngữ cảnh của câu phía trên

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -113,21 +133,21 @@
-||page|3|number|Display employee information page|
+||first_name|bien|string|Employee's first name|
+||last_name|danghuu|string|Employee's last name|
+||date_of_birth_from|1996-02-15|string|Birthdate of Employee|
```
cũng là dạng sở hữu nhưng sao discription khác nhau

huubien replied:
Em đã sửa lại Employee's Birthdate.

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -113,21 +133,21 @@
+||date_of_birth_to|1998-12-30|string|Birthdate of Employee|
+||salary_min|1000|number|Min salary of Employee|
+||salary_max|50000|number|Max salary of Employee|
+||page|3|number|Display Employee information page|
```
ghi cái gì vậy? làm gì có chuyện hiển thị ở đây?

huubien replied:
Dạ không có Display. Em đã sửa lại: The number of current page is 3

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -177,20 +196,20 @@
-|404|not_found|Not found employee.|Not found employee information. |
+|400|invalid_date_of_birth_from|Date of birth is invalid.|The user enters a birthday that is not a valid date format.|
+|400|invalid_date_of_birth_to|Date of birth is invalid.|The user enters a birthday that is not a valid date format.|
+|400|invalid_salary_min|Employee salary is invalid.|Input argument of type 'char'|
```
cũng là description của error tại sao 2 cách ghi khác nhau vậy?

huubien replied:
Em đã sửa lại

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -177,20 +196,20 @@
-  "code": "invalid_date_of_birth_start",
-  "message": "Date Of Birth is invalid."
+  "code": "invalid_date_of_birth_from",
+  "message": "Date of birth is invalid."
```
`Date format is invalid.` thì sao?

huubien replied:
Em đã tìm hiểu thấy hợp lý và được sử dụng phổ biến trong error message. Em đã sửa lại Date format is invalid.

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -288,19 +334,30 @@
+|○|first_name|bien|string|First name|
+|○|last_name|danghuu|string|Last name|
+|○|date_of_birth|1996-02-15|string|Birthdate|
+|○|salary|50000|number|Employee's salary|
```
tại sao chổ này required?

huubien replied:
Lương có thể bắt buộc hoặc không bắt buộc nhưng trong app của e nếu không nhập lương sẽ không thực thi được các nghiệp vụ tính lương.

lannt-xyz replied:
vậy thì tại sao chức năng đăng ký ko bắt buộc nhập mà chức năng update lại bắt buộc?

huubien replied:
Vâng em sửa thiếu. Em đã bổ sung bắt buộc nhập salary ở chức năng đăng ký.

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -311,43 +368,27 @@
-|400|invalid_salary|Request employee salary is invalid.|The requested Employee salary is contains non-string characters. |
-|400|required_id|Must not be empty.|Return if employee id is not specified. |
-|404|not_found|Not found employee.|Not found employee information. |
+|400|required_id|Must not be empty.|Employee id is not specified.|
```
`required_id`, `invalid_employee_id` tại sao lại có sự khác biệt như vậy?

huubien replied:
Vâng em sửa thiếu. Em đã chỉnh sửa lại required_employee_id

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -311,43 +368,27 @@
-|404|not_found|Not found employee.|Not found employee information. |
+|400|required_id|Must not be empty.|Employee id is not specified.|
+|400|invalid_employee_id|Employee ID is invalid.|The requested Employee ID is contains Non-Nummeric characters.|
+|400|invalid_date_of_birth|Date of birth is invalid.|The user enters a birthday that is not a valid date format.|
```
`The user ` là thằng quái nào vậy? linh tinh

lannt-xyz replied:
cũng là description cho error đối với date format, mà lại ghi mỗi nơi mỗi khác?

huubien replied:
Vâng em đã chỉnh sửa lại.

lannt-xyz commented on employee-management/interface-document.md:
```diff
@@ -358,11 +399,11 @@
     {
       "code": "invalid_date_of_birth",
-      "message": "Date Of Birth is invalid."
+      "message": "Date of birth is invalid."
```
chổ này sao sửa lại vậy? ko phải ban đầu nó đã đúng rồi sao/

huubien replied:
Em nghĩ chỉ viết hoa chữ cái đầu, Em đã chỉnh sửa Date Of Birth.

lannt-xyz commented on employee-management/interface-document.md:
````diff
@@ -398,26 +445,39 @@
+
+```json
+{
+  "errors": [
````
vấn đề này nói nhiều lần rồi nha.
sao có thể cùng lúc xảy ra 2 trường hợp như bên dưới. -> vớ vẩn

huubien replied:
Em đã chỉnh sửa lại trường hợp lỗi này

huubien committed:
Change Description, Repair Sample Error Response'

Change 'Description' at all functions

Repair 'Sample Error Response' at Delete, Detail Employee functions

huubien committed:
repair 'Request Body'

add Required for salary in the registration function

huubien committed:
"Source code Employee_API"

huubien committed:
"Update source code Employee_API"
